### PR TITLE
ENT-277: Fix IntegrityError getting raised for existing enterprise users

### PR DESCRIPTION
**Pull request by zubair-arbi**
ENT-277
@saleem-latif @asadiqbal08 @mattdrayer 

**Description:** Fixed `IntegrityError` getting raised for existing enterprise users (manually added/linked from enterprise django admin `Manage Learners`)  when the data sharing consent is disabled for the related enterprise.

**JIRA:** [ENT-277](https://openedx.atlassian.net/browse/ENT-277)

**Dependencies:** N/A

**Merge deadline:** 7 April 2017

**Installation instructions:** 
1. Setup edx-platform (master) with enterprise branch set to this.
2. Set up a TestShib SAML integration with the instructions [found here](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/tpa/tpa_SAML_SP.html).
3. Create an EnterpriseCustomer in the Django admin.

**Testing instructions:**
1. Link the EnterpriseCustomer to the TestShib SAML integration
2. Verify that data sharing consent is set to disabled on the EnterpriseCustomer
3. Link learner account with enterprise through manage learner page in Django admin
4. In an incognito window, browse to the LMS registration page
5. Clicking on the `TestShib` button will redirect learner to the `TestShib` auth page (third party)
6. After entering credentials on `TestShib` auth page it will redirect learner back to LMS logistration page.
7. Provide the credentials of already linked learner with enterprise(that linked in step-3) in login form.
8. Press login button and user will be redirected to LMS dashboard.
9. Observe that your SAML user is linked as an Enterprise Learner and also a new social auth entry is added in `Python Social Auth` for that user.

**Merge checklist:**

- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [x] Commits are (reasonably) squashed
- [ ] Translations are updated
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: enterprise/tpa_pipeline.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def redirect_to_consent():
     if not enterprise_customer.requests_data_sharing_consent:
         # This enterprise customer attached to this pipeline element does not request data sharing consent;
         # proceed with the creation of a link between the user and the enterprise customer, then exit.
-        EnterpriseCustomerUser.objects.create(
+        EnterpriseCustomerUser.objects.get_or_create(
```
Comment (mattdrayer) [inline enterprise/tpa_pipeline.py]: Wow, is that all it was?  Good catch!  😄

**Comment by mattdrayer** [comment]
LGTM -- 👍

File: enterprise/tpa_pipeline.py (hunk 1 above)
Comment (zubair-arbi) [inline enterprise/tpa_pipeline.py]: @mattdrayer yeah. Fixing was quick but correctly configuring tpa-saml on my local machine took 2 days and thanks to @saleem-latif I got it working today.  :stuck_out_tongue_closed_eyes:

Comment (mattdrayer) [inline enterprise/tpa_pipeline.py]: Think of how much you have learned!  😉

**Pull request merged**